rcljava/src/test/java/org/ros2/rcljava/node/NodeParametersTest.java

```diff
@@ -0,0 +1,383 @@
+import org.junit.Before;
+import org.junit.Test;
+
+import org.ros2.rcljava.RCLJava;
+import org.ros2.rcljava.parameters.ParameterCallback;
+import org.ros2.rcljava.parameters.ParameterType;
+import org.ros2.rcljava.parameters.ParameterVariant;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+public class NodeParametersTest {
+  private Node node;
+  private boolean callbackCalled = false;
+
+  @BeforeClass
+  public static void setupOnce() throws Exception {
+    // Just to quiet down warnings
+    org.apache.log4j.BasicConfigurator.configure();
+
+    RCLJava.rclJavaInit();
+  }
+
+  @AfterClass
+  public static void tearDownOnce() {
+    RCLJava.shutdown();
+  }
+
+  @Before
+  public void setUp() {
+    node = RCLJava.createNode("test_node");
+  }
+
+  @After
+  public void tearDown() {
+    node.dispose();
+  }
+
+  @Test
+  public final void testDeclareParameter() {
+    ParameterVariant p = new ParameterVariant("foo", true);
+    node.declareParameter(p);
+    assertTrue(node.hasParameter("foo"));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public final void testDeclareParameterTwice() {
+    ParameterVariant p = new ParameterVariant("foo", true);
+    node.declareParameter(p);
+    node.declareParameter(p);
+  }
+
+  @Test
+  public final void testDeclareParameters() {
+    List<ParameterVariant> plist = new ArrayList<ParameterVariant>();
+    List<rcl_interfaces.msg.ParameterDescriptor> pdlist = new ArrayList<rcl_interfaces.msg.ParameterDescriptor>();
+
+    plist.add(new ParameterVariant("foo", true));
+    pdlist.add(new rcl_interfaces.msg.ParameterDescriptor());
+
+    node.declareParameters(plist, pdlist);
+    assertTrue(node.hasParameter("foo"));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public final void testDeclareParametersBadListLength() {
+    List<ParameterVariant> plist = new ArrayList<ParameterVariant>();
+    List<rcl_interfaces.msg.ParameterDescriptor> pdlist = new ArrayList<rcl_interfaces.msg.ParameterDescriptor>();
+
+    plist.add(new ParameterVariant("foo", true));
+
+    node.declareParameters(plist, pdlist);
+  }
+
+  @Test
+  public final void testUndeclareParameter() {
+    ParameterVariant p = new ParameterVariant("foo", true);
+    node.declareParameter(p);
+    assertTrue(node.hasParameter("foo"));
+    node.undeclareParameter("foo");
+    assertFalse(node.hasParameter("foo"));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public final void testUndeclareParameterDoesNotExist() {
+    node.undeclareParameter("foo");
+  }
+
+  @Test
+  public final void testGetParameters() {
+    ParameterVariant p = new ParameterVariant("foo", true);
+    node.declareParameter(p);
+    assertTrue(node.hasParameter("foo"));
+    List<ParameterVariant> results = node.getParameters(new ArrayList<String>(Arrays.asList("foo")));
+    assertEquals(1, results.size());
+    assertEquals("foo", results.get(0).getName());
```

Comment:
Can we test getting the value too?

Reply:
Good idea.  Done in 36daa72840b68189f0c6d7d763440c2395ec0d46